### ISSUE TITLE
Phoenix connector throw unknown connector error when table has unsupported column type

### ISSUE THREAD

**Issue by rongyousu** [prestosql/presto#2416]
Perhaps, this issue was introduced in https://github.com/prestosql/presto/pull/1182

**Comment by ebyhr** [comment]
Could you modify the issue title (what's the actual issue)?

**Comment by rongyousu** [comment]
2020-01-06T15:19:53.439Z	DEBUG	20200106_151953_00055_5bir6.2.0-0-114	io.prestosql.plugin.jdbc.BaseJdbcClient	Mapping data type of 'dbcenter.tb_user_info' column 'ID_CARD': JdbcTypeHandle{jdbcType=12, jdbcTypeName=VARCHAR, columnSize=0, decimalDigits=0} mapped to Optional[ColumnMapping{type=varchar}]
2020-01-06T15:19:53.439Z	DEBUG	20200106_151953_00055_5bir6.2.0-0-114	io.prestosql.plugin.jdbc.BaseJdbcClient	Mapping data type of 'dbcenter.tb_user_info' column 'IS_REAL_NAME_LOCKED': JdbcTypeHandle{jdbcType=-5, jdbcTypeName=BIGINT, columnSize=0, decimalDigits=0} mapped to Optional[ColumnMapping{type=bigint}]
2020-01-06T15:19:53.439Z	DEBUG	20200106_151953_00055_5bir6.2.0-0-114	io.prestosql.plugin.jdbc.BaseJdbcClient	Mapping data type of 'dbcenter.tb_user_info' column 'REAL_NAME_LOCKED_TIME': JdbcTypeHandle{jdbcType=93, jdbcTypeName=TIMESTAMP, columnSize=0, decimalDigits=0} mapped to Optional.empty
2020-01-06T15:19:53.445Z	DEBUG	task-notification-0	io.prestosql.execution.TaskStateMachine	Task 20200106_151953_00055_5bir6.2.0 is FAILED
2020-01-06T15:19:53.439Z	ERROR	SplitRunner-5-114	io.prestosql.execution.executor.TaskExecutor	Error processing Split 20200106_151953_00055_5bir6.2.0-0 io.prestosql.connector.informationschema.InformationSchemaSplit@1eb76527 (start = 2.100917789501379E9, wall = 75 ms, cpu = 0 ms, wait = 0 ms, calls = 1): INVALID_SESSION_PROPERTY: Unknown connector phoenix
io.prestosql.spi.PrestoException: Unknown connector phoenix
	at io.prestosql.metadata.SessionPropertyManager.getConnectorSessionPropertyMetadata(SessionPropertyManager.java:121)
	at io.prestosql.metadata.SessionPropertyManager.decodeCatalogPropertyValue(SessionPropertyManager.java:182)
	at io.prestosql.FullConnectorSession.getProperty(FullConnectorSession.java:132)
	at io.prestosql.plugin.jdbc.BaseJdbcPropertiesProvider.getUnsupportedTypeHandling(BaseJdbcPropertiesProvider.java:53)
	at io.prestosql.plugin.jdbc.BaseJdbcClient.getColumns(BaseJdbcClient.java:275)
	at io.prestosql.plugin.phoenix.PhoenixMetadata.getTableMetadata(PhoenixMetadata.java:143)
	at io.prestosql.plugin.phoenix.PhoenixMetadata.getTableMetadata(PhoenixMetadata.java:137)
	at io.prestosql.plugin.phoenix.PhoenixMetadata.lambda$listTableColumns$15(PhoenixMetadata.java:452)
	at java.util.Optional.ifPresent(Optional.java:159)
	at io.prestosql.plugin.phoenix.PhoenixMetadata.listTableColumns(PhoenixMetadata.java:452)
	at io.prestosql.spi.connector.classloader.ClassLoaderSafeConnectorMetadata.listTableColumns(ClassLoaderSafeConnectorMetadata.java:258)
	at io.prestosql.metadata.MetadataManager.listTableColumns(MetadataManager.java:566)
	at io.prestosql.metadata.MetadataListing.listTableColumns(MetadataListing.java:93)
	at io.prestosql.connector.informationschema.InformationSchemaPageSource.addColumnsRecords(InformationSchemaPageSource.java:237)
	at io.prestosql.connector.informationschema.InformationSchemaPageSource.buildPages(InformationSchemaPageSource.java:205)
	at io.prestosql.connector.informationschema.InformationSchemaPageSource.getNextPage(InformationSchemaPageSource.java:171)
	at io.prestosql.operator.ScanFilterAndProjectOperator$ConnectorPageSourceToPages.process(ScanFilterAndProjectOperator.java:362)
	at io.prestosql.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:372)
	at io.prestosql.operator.WorkProcessorUtils.getNextState(WorkProcessorUtils.java:221)
	at io.prestosql.operator.WorkProcessorUtils.access$000(WorkProcessorUtils.java:37)
	at io.prestosql.operator.WorkProcessorUtils$YieldingProcess.process(WorkProcessorUtils.java:181)
	at io.prestosql.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:372)
	at io.prestosql.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:306)
	at io.prestosql.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:372)
	at io.prestosql.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:306)
	at io.prestosql.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:372)
	at io.prestosql.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:306)
	at io.prestosql.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:372)
	at io.prestosql.operator.WorkProcessorUtils.getNextState(WorkProcessorUtils.java:221)
	at io.prestosql.operator.WorkProcessorUtils.lambda$processStateMonitor$2(WorkProcessorUtils.java:200)
	at io.prestosql.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:372)
	at io.prestosql.operator.WorkProcessorUtils.lambda$flatten$6(WorkProcessorUtils.java:277)
	at io.prestosql.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:319)
	at io.prestosql.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:372)
	at io.prestosql.operator.WorkProcessorUtils$3.process(WorkProcessorUtils.java:306)
	at io.prestosql.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:372)
	at io.prestosql.operator.WorkProcessorUtils.getNextState(WorkProcessorUtils.java:221)
	at io.prestosql.operator.WorkProcessorUtils.lambda$processStateMonitor$2(WorkProcessorUtils.java:200)
	at io.prestosql.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:372)
	at io.prestosql.operator.WorkProcessorUtils.getNextState(WorkProcessorUtils.java:221)
	at io.prestosql.operator.WorkProcessorUtils.lambda$finishWhen$3(WorkProcessorUtils.java:215)
	at io.prestosql.operator.WorkProcessorUtils$ProcessWorkProcessor.process(WorkProcessorUtils.java:372)
	at io.prestosql.operator.WorkProcessorSourceOperatorAdapter.getOutput(WorkProcessorSourceOperatorAdapter.java:148)
	at io.prestosql.operator.Driver.processInternal(Driver.java:379)
	at io.prestosql.operator.Driver.lambda$processFor$8(Driver.java:283)
	at io.prestosql.operator.Driver.tryWithLock(Driver.java:675)
	at io.prestosql.operator.Driver.processFor(Driver.java:276)
	at io.prestosql.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1075)
	at io.prestosql.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:163)
	at io.prestosql.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:484)
	at io.prestosql.$gen.Presto_327____20200106_121702_2.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

**Comment by rongyousu** [comment]
the version 327 is not better than 324 , because  the  timestamp column

**Comment by rongyousu** [comment]
https://github.com/prestosql/presto/blob/370a832d8730ca6e82932fe8b678a1b3d7a74d6d/presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixClient.java ，the type not code


 public Optional<ColumnMapping> toPrestoType(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
    {
        switch (typeHandle.getJdbcType()) {
           ......
            // TODO add support for TIMESTAMP after Phoenix adds support for LocalDateTime
            case TIMESTAMP:
            case TIME_WITH_TIMEZONE:
            case TIMESTAMP_WITH_TIMEZONE:
                return Optional.empty();
            case FLOAT:

**Comment by ebyhr** [comment]
It seems the change for `unsupported_type_handling` affected it.
https://prestosql.io/docs/current/release/release-327.html

**Comment by ebyhr** [comment]
Steps to reproduce:
Add this test to TestPhoenixIntegrationSmokeTest and run 
```java
    @Test
    public void testTimestamp()
            throws Exception
    {
        executeInPhoenix("CREATE TABLE tpch.test_timestamp (pk bigint primary key, val1 timestamp)");
        executeInPhoenix("UPSERT INTO tpch.test_timestamp (pk, val1) VALUES (1, timestamp '2002-05-30T09:30:10.5')");
        assertQuery("SELECT pk FROM test_timestamp", "SELECT 1");
    }

```

**Comment by rongyousu** [comment]
hi,,i also find  the phoenix.config   can't use "unsupported-type-handling=CONVERT_TO_VARCHAR"  and  "jdbc-types-mapped-to-varchar"   

1) Configuration property 'unsupported-type-handling' was not used
  at io.airlift.bootstrap.Bootstrap.lambda$initialize$2(Bootstrap.java:238)

1 error
com.google.inject.CreationException: Unable to create injector, see the following errors:

next version is ok?  ( add  unsupported-type-handling=CONVERT_TO_VARCHAR  )

**Comment by ebyhr** [comment]
It is session property (not static config) and setting a session property will be failed with the same cause. While I already created the PR (#2424) to fix the regression, there is no existing PR to support the property for now.

**Comment by kokosing** [comment]
Fixed via #2427
